modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java

```diff
@@ -278,13 +285,33 @@ public Response getMediaPackage(@PathParam("mediaPackageID") final String mediaP
               responseCode = SC_INTERNAL_SERVER_ERROR)})
   public Response getAsset(@PathParam("mediaPackageID") final String mediaPackageID,
                            @PathParam("mediaPackageElementID") final String mediaPackageElementID,
-                           @PathParam("version") final String version) {
+                           @PathParam("version") final String version,
+                           @HeaderParam("If-None-Match") String ifNoneMatch) {
+
     try {
       for (final Version v : getAssetManager().toVersion(version)) {
         for (Asset asset : getAssetManager().getAsset(v, mediaPackageID, mediaPackageElementID)) {
+
+          if (StringUtils.isNotBlank(ifNoneMatch)) {
+            Checksum checksum = asset.getChecksum();
+
+            if (checksum != null & checksum.getType().equals(ChecksumType.DEFAULT_TYPE)) {
```

Comment:
Single `&` here, which I suspect is not the goal.  Also, we're checking the type, under what circumstances can this *not* be `DEFAULT_TYPE`?  Why do we care what the type is if we're getting the checksum and checking it against the `ifNoneMatch` field?

Reply:
I guess right now it can't, I mainly wanted to make sure that both checksums were created by the same algorithm even in the future (although for them to match when that isn't the case is not very probable..)

Reply:
Fixed the &.